hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -2736,8 +2738,12 @@ public void testClusteringCommitInPresenceOfInflightCommit() throws Exception {
     assertThrows(HoodieClusteringException.class, () -> clusteringWriteClient.cluster(clusteringCommitTime, true));
 
     // Do a rollback on the replacecommit that is failed
-    clusteringWriteClient.rollback(clusteringCommitTime);
+    // clusteringWriteClient.rollback(clusteringCommitTime);
```

Comment:
as per my understanding, there should not be a need to rollback the clustering commit explicitly.. it should be done automatically if a conflict is detected by the new strategy. But, both replacecommit inflight and requested remain in the timeline.

Reply:
Actually, rollback of these clustering commits is handled separately, so it will leave out the inflight in the timeline. We use SparkAllowUpdateStrategy so only those cases you are allowed to use IngestionPrimaryWriterBasedConflictResolutionStrategy, we are using couple of approaches to clean these inflights, one by explicitly assigning a rollback for the failed commit and another approach is by including replacecommits as part of rollbackFailedWrites that way ingestion takes care of clearing them.
I think we need to make immutable nature of clustering commits as a table property i.e. store it in hoodie.properties. That way ingestion knows whether a clustering commits can be rolled back or not and accordingly it can either use SparkRejectUpdateStrategy or SparkAllowUpdateStrategy implementations and cleanup can. be done separately.